File: Python/0050_pow_x_n.py
# 2020-08-05 14:14:54

class Solution:
    def myPow(self, x: float, n: int) -> float:
        N = abs(n)
        res = 1.0
        while N != 0:
            if (N&1) == 1:
                res *= x
            x *= x
            N >>= 1
        return res if n > 0 else 1 / res

File: TypeScript/0050-pow-x-n.ts
// 2020-08-05 16:15:03

function myPow(x: number, n: number): number {
  let N = Math.abs(n);
  let res = 1.0
  while (N != 0) {
    if (N % 2 == 1) {
      res *= x;
    }
    x *= x;
    N >>>= 1;
  }
  return n < 0 ? 1.0 / res : res;
};
